Source/FlowEditor/Public/Asset/FlowDebuggerSubsystem.h

```diff
@@ -34,4 +102,85 @@ class FLOWEDITOR_API UFlowDebuggerSubsystem : public UEditorSubsystem
+	static void RemoveTrait(const UEdGraphPin* OwnerPin, EFlowTraitType Type);
+
+	/** Removes any OwnerNode's trait that matches the provided Predicate */
+	static void RemoveNodeTraitByPredicate(const UFlowGraphNode* OwnerNode, const TFunctionRef<bool(const FFlowDebugTrait&)> Predicate);
+	/** Removes any pin trait of provided OwnerNode that matches the provided Predicate */
+	static void RemovePinTraitByPredicate(const UFlowGraphNode* OwnerNode, const TFunctionRef<bool(const FFlowDebugTrait&)> Predicate);
+	/** Removes any OwnerPin's trait that matches the provided Predicate */
+	static void RemovePinTraitByPredicate(const UEdGraphPin* OwnerPin, const TFunctionRef<bool(const FFlowDebugTrait&)> Predicate);
+	
+	/** Clears all node traits for provided OwnerNode */
+	static void ClearNodeTraits(const UFlowGraphNode* OwnerNode);
+	/** Clears all pin traits for provided OwnerNode */
+	static void ClearPinTraits(const UFlowGraphNode* OwnerNode);
+	/** Clears all traits for provided OwnerPin */
+	static void ClearPinTraits(const UEdGraphPin* OwnerPin);
+	/** Removes stale pin traits for provided OwnerNode. Pin list can be changed after node reconstructing and traits for removed pins can stay in PerNodeSettings.
+	 * There is no need to clean node's traits here, cause all node events can be processed right away and there will be no stale nodes in PerNodeSettings */
+	static void CleanupTraits(const UFlowGraphNode* OwnerNode);
+
+	/** Finds OwnerNode's trait with provided Type
+	 * returns null if there are no OwnerNode's trait with provided Type */
+	static FFlowDebugTrait* FindTrait(const UFlowGraphNode* OwnerNode, EFlowTraitType Type);
+	/** Finds OwnerPin's trait with Type
+	 * returns null if there are no OwnerPin's trait with provided Type */
+	static FFlowDebugTrait* FindTrait(const UEdGraphPin* OwnerPin, EFlowTraitType Type);
+
+	/** Sets or clears the enabled flag for the OwnerNode's trait with provided Type */
+	static void SetTraitEnabled(const UFlowGraphNode* OwnerNode, EFlowTraitType Type, bool bIsEnabled);
+	/** Sets or clears the enabled flag for the OwnerPin's trait with provided Type */
+	static void SetTraitEnabled(const UEdGraphPin* OwnerPin, EFlowTraitType Type, bool bIsEnabled);
+
+	/** Returns the enabled flag for the OwnerNode's trait with provided Type */
+	static bool IsTraitEnabled(const UFlowGraphNode* OwnerNode, EFlowTraitType Type);
+	/** Returns the enabled flag for the OwnerPin's trait with provided Type */
+	static bool IsTraitEnabled(const UEdGraphPin* OwnerPin, EFlowTraitType Type);
+
+	/** Creates or removes OwnerNode's trait with provided Type */
+	static void ToggleTrait(const UFlowGraphNode* OwnerNode, EFlowTraitType Type);
+	/** Creates or removes OwnerPin's trait with provided Type */
+	static void ToggleTrait(const UEdGraphPin* OwnerPin, EFlowTraitType Type);
+
+	/** Sets the hit flag for the all OwnerNode's traits */
+	static TArray<EFlowTraitType> SetAllTraitsHit(const UFlowGraphNode* OwnerNode, bool bHit);
```

Comment:
Maybe it would be better to return bitmask here? I doubt that in future there will be over 32/64 trait types